crypto-primitives/src/commitment/pedersen/constraints.rs

```diff
@@ -142,6 +152,16 @@ where
     G: Group,
     ConstraintF: PrimeField,
 {
+    fn alloc_constant<T, CS: ConstraintSystem<ConstraintF>>(
+        cs: CS,
+        val: T,
+    ) -> Result<Self, SynthesisError>
+    where
+        T: Borrow<PedersenRandomness<G>>,
+    {
+        Self::alloc(cs, || Ok(val))
```

Comment:
Should this call `alloc_constant` of the underlying bytes? Maybe something like:
```suggestion
		let temp = value_gen()?;
        let mut result = vec![];
        for (i, byte) in to_bytes![temp.borrow().0].unwrap().into_iter().enumerate() {
        	let cur = UInt8::alloc_constant(cs.ns(|| format!("byte {}", i)), || Ok(byte))?;
        	result_bytes.push(cur);
        }
        Ok(PedersenRandomnessGadget(result))
```

Reply:
(Also similarly for the Bowe-Hopwood-Pedersen hash, right?

Reply:
You're right, yeah!

Reply:
I think this comment applies only to the Pedersen commitment, and the first comment to both Pedersen and BH hashes.